### ISSUE TITLE
after upgrade superset 2.0 cant render a chart with event annotation on dashboard 

### ISSUE THREAD

**Issue by oleg-savko** [apache/superset#20845]
After upgrading superset to 2.0, it cant render a chart with Event Annotation on dashboard and return:

```
Unexpected error
An error occurred while rendering the visualization: TypeError: Cannot use 'in' operator to search for 'columns' in undefined

```

![Screenshot 2022-07-25 at 13 37 28](https://user-images.githubusercontent.com/14272376/180758040-afaaccbe-b8ab-4194-a47f-8d29971e67c4.png)

![Screenshot 2022-07-25 at 13 43 23](https://user-images.githubusercontent.com/14272376/180759248-c442ec17-463c-4892-bd21-33b56e6ca0a9.png)

If remove annotation chart render properly.

Expected: chart with event annotation worked as before.

**Comment by sowo** [comment]
I confirm this issue after upgrading from from 1.5.1 to 2.0, annotations cannot be rendered. In my case, I use an annotation layer which results in the same error as shown above.

when hiding the annotation, the charts render just fine.

**Comment by alexey-kasiyanik** [comment]
Yes, have faced this issue as well :(

**Comment by xmnlab** [comment]
hi everyone! just a friendly ping here to check if there is any update about this issue

**Comment by joaocmd93** [comment]
Hello, any updates on this issue?

**Comment by 1-Tyler** [comment]
Workaround: 
On the chart, if you remove the annotations and re-add them back, the error should go away and will load correctly. 
Or you can click 'apply' and save the annotation and it will work.

The GET call to 'api/v1/chart/CHART_ID' is returning error 404.

**Comment by joaocmd93** [comment]
> Workaround: On the chart, if you remove the annotations and re-add them back, the error should go away and will load correctly. Or you can click 'apply' and save the annotation and it will work.
> 
> The GET call to 'api/v1/chart/CHART_ID' is returning error 404.

For me this only works on the 1st load after reapplying the annotation.
If I save and try to load again the error continues.

**Comment by 1-Tyler** [comment]
Ah you are right @joaocmd93 , I am seeing that behavior as well unfortunately.

**Comment by bonamim** [comment]
Facing the same error here 😭

**Comment by zhaorui2022** [comment]
We also did an upgrade and users are facing this issue with some critical dashboards. Is there any solution to it yet? If no, we have to roll back to our previous version.

**Comment by xmnlab** [comment]
nice!

**Comment by rusackas** [comment]
I believe this was fixed by #20651, but this issue/ticket was never actually closed. Holler if this is still an issue in 3.x, and we can discuss!
